### PR TITLE
ADBDEV-6631: Don't use fatal on gprestore helper errors

### PR DESCRIPTION
**Pull request by RekGRpth**
Don't use fatal on gprestore helper errors

The main gprestore process panics if an error occurs on the helper and it
creates an error file. Since the creation of an error file by the helper is a
regular situation, it is enough to cancel the context to terminate all parallel
goroutines.

### CODE REVIEW

**Comment by RekGRpth** [comment]
fixed by https://github.com/arenadata/gpbackup/pull/111

File: utils/agent_remote.go [hunk 1]
```diff
@@ -346,7 +347,7 @@ func CheckAgentErrorsOnSegments(c *cluster.Cluster, fpInfo filepath.FilePathInfo
 	}
 	if numErrors > 0 {
 		helperLogName := fpInfo.GetHelperLogPath()
-		return errors.Errorf("Encountered errors with %d helper agent(s).  See %s for a complete list of segments with errors, and see %s on the corresponding hosts for detailed error messages.",
+		return errors.Wrapf(AgentErr, "Encountered errors with %d helper agent(s).  See %s for a complete list of segments with errors, and see %s on the corresponding hosts for detailed error messages.",
```
Comment (andr-sokolov) [inline utils/agent_remote.go:350]: Wrapf returns an error with a stack trace. Do we need  stack trace? Can we use another function, which doesn't collect  stack trace?

Comment (RekGRpth) [reply utils/agent_remote.go:350]: > Wrapf returns an error with a stack trace.

like `Errorf` does

> Can we use another function, which doesn't collect stack trace?

`WithMessagef` annotates `err` with the `format` specifier



> Do we need stack trace?

Maybe we need in future

**Comment by andr-sokolov** [comment]
I suggest to rewrite the last sentence: "Since the creation of an error file by the helper is a regular situation, it is enough to cancel the context to terminate all parallel goroutines."

**Comment by RekGRpth** [comment]
> I suggest to rewrite the last sentence: "Since the creation of an error file by the helper is a regular situation, it is enough to cancel the context to terminate all parallel goroutines."

rewrited

**Review by andr-sokolov — APPROVED** [review]

**Review by dnskvlnk — APPROVED** [review]

**Pull request merged**